src/avt/Blueprint/avtConduitBlueprintDataAdaptor.C

```diff
@@ -1138,6 +1139,86 @@ void vtkUnstructuredToNode(Node &node,
   }
 }
 
+
+// ****************************************************************************
+void vtkPolyDataToNode(Node &node, 
+                       vtkPolyData *grid,
+                       int /*ndims*/)
+{
+  const int npts = grid->GetNumberOfPoints();
+  const int nzones = grid->GetNumberOfCells();
+
+  if(nzones == 0)
+  {
+    AVT_CONDUIT_BP_EXCEPTION1(InvalidVariableException,
+                              "PolyData grids must have at least one "
+                              "zone");
+  }
+
+  bool single_shape = true;
+  int cell_type = grid->GetCell(0)->GetCellType();
+  for(int i = 1; i < nzones; ++i)
+  {
+    vtkCell *cell = grid->GetCell(i);
+    if(cell->GetCellType() != cell_type)
+    {
+      single_shape = false;
+      break;
+    }
+  }
+
+  if(!single_shape)
+  {
+    AVT_CONDUIT_BP_EXCEPTION1(InvalidVariableException,
+                              "Only PolyData grids with a single cell type"
+                              " are currently supported");
+  }
+
+  const std::string shape_name = VTKCellTypeToElementShapeName(cell_type);
+  if(shape_name == "")
+  {
+    AVT_CONDUIT_BP_EXCEPTION1(InvalidVariableException,
+                              "Unsupported vtk cell type "<<cell_type);
+  }
+
+  node["shape"] = shape_name;
+
+  const int cell_points = grid->GetCell(0)->GetNumberOfPoints();
+  node["connectivity"].set(DataType::int32(cell_points * nzones));
+  conduit::int32 *conn = node["connectivity"].value();
+  // vtk connectivity is in the form npts, p0, p1,..
+  // and we need p0, p1, .. so just iterate and copy
+  if(cell_type != VTK_VOXEL)
+  {
+    for(int i = 0; i < nzones; ++i)
+    {
+      vtkCell *cell = grid->GetCell(i);
+      const int offset = i * cell_points;
+      for(int c = 0; c < cell_points; ++c)
+      {
+        conn[offset + c]  = cell->GetPointId(c);
+      }
+    }
+  }
+  else
+  {
+    // We need to reorder the voxel indices to be a hex
+    int reorder[8] = {0, 1, 3, 2, 4, 5, 7, 6};
```

Comment:
My only thought is that I wish this was somehow lifted out of this function, since if anything were to change in the future we would have to track this translation down here, or in case anyone else would want to use this translation. Probably things won't change though.

Reply:
@cyrush just curious but do you envision a future where Conduit/Blueprint doesn't rely on these being hard coded into a producer or cosumer and instead passes the permutation *with* the data itself?

Reply:
Right, I figured that the orderings for VTK and Conduit are very unlikely to change.

But it might be nice if each time someone wanted to translate between VTK voxel and hex they didn't have to look it up or find where it was done elsewhere. I like the idea of a const static array but maybe you're right that that would be obtuse or unnecessary.